### DISCUSSION TITLE
3D viewport and Thumbnail Generation/Customization for Industry Standard Formats like USD, FBX , OBJ , GLB etc

### DISCUSSION THREAD

**Discussion by Da3n0n** [spacedriveapp/spacedrive#2454] [Feature Request]
its exactly what it says this would allow users to preview and customize thumbnails for 3d files of various formats. 

Whats Actually Needed ?

- Quick View All Industry Standard Formats like USD , FBX , OBJ ,GLB etc.
- allow to change/save Up Axis/vector on a per file basis. or just save state from last view in viewport would be alternative  .
- Allow to save any view in viewport as thumbnail this will allow users to make readable thumbnails even is the model is not rotated correctly an alternative here is to allow copy from clipboard to set as thumbnail . then users can simply screenshot the viewport and set as thumnail.

Nice to haves.

- HDRI Preview when previewing HDRI this would open a basic 3D viewport with the HDRI set as the Dome with few shader balls (metal, Glass, etc , this would allow users to get a lighting preview of how the HDRI is gonna look in 3D Before importing . look at examples of when you open HDRI in Eagle.cool Digital asset manager

**Comment by ketonik** [comment]
(Personally i've no need to change axis in a preview of a file in a file manager, wtff??

Anyway, Thumbnail Preview for the following file formats would be most welcome:
dwg, dxf dwf, dwfx
Inventor ipt iam
Revit rfa rvt
Sketchup skp
Collada dae
Rhino 3dm
ifc
Standard for Exchange Product Data stp, step, stpz
Parasolid x_t
Siemens NX prt
Siemens Solid Edge par asm psm
Solidworks sldprt sldasm

**Reply by anscarlett** [reply]
"Personally i've no need to change axis in a preview of a file in a file manager, wtff?"

I assume they mean the axis that is considered as up, so models all appear to be standing on a ground plane. It's common for models to use either the Y or Z axis as this. That, I believe, is what.

**Comment by HeavenVolkoff** [comment]
I created an issue for adding support for generating thumbnails of 3D models:
https://github.com/spacedriveapp/spacedrive/issues/2784
